eskip/eskip.go

```diff
@@ -44,13 +53,28 @@ type parsedRoute struct {
 	backend  string
 }
 
+// A CustomPredicate object represents a parsed, in-memory, route matching predicate
+// that is defined by extensions.
+type CustomPredicate struct {
```

Comment:
I also vote for a simplified Predicate. I thought this to be new code, are there already any dependencies on `CustomPredicate`?